src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb

```diff
@@ -664,6 +664,21 @@ Namespace Microsoft.VisualStudio.Editors.AppDesCommon
 
                 TelemetryService.DefaultSession.PostEvent(userTask)
             End Sub
+
+            Private Const BinaryFormatterEventName As String = "vs/projectsystem/appdesigner/binaryformatter"
+            Public Enum BinaryFormatterType
+                Serialize = 0
+                Deserialize = 1
+            End Enum
+
+            Public Shared Sub LogBinaryFormatterEvent(functionName As String, className As String, type As BinaryFormatterType)
```

Comment:
You can use `CallerMemberName` so that call sites don't have to pass their function name. The compiler will populate the argument automatically. There's no `CallerTypeName` unfortunately though.

```suggestion
            Public Shared Sub LogBinaryFormatterEvent(className As String, type As BinaryFormatterType, <System.Runtime.CompilerServices.CallerMemberName> Optional functionName As String = Nothing)
```

Reply:
I avoided this originally as I thought it used reflection. Looked it up; it does not. https://stackoverflow.com/a/22580624/294804

Reply:
Done.